nsqd/topic.go

```diff
@@ -263,34 +269,49 @@ func (t *Topic) messagePump() {
 			goto exit
 		}
 
-		for i, channel := range chans {
```

Comment:
and then this loop happens while some other process could be trying to change the set of channels. I'm not sure that's a problem, but putting this under the topic's lock will help make channel operations more atomic.